core/templates/domain/blog/blog-dashboard-backend-api.service.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to get data for to a blog dashboard from backend.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BlogPostSummaryBackendDict, BlogPostSummary } from 'domain/blog/blog-post-summary.model';
+import { BlogDashboardPageConstants } from 'pages/blog-dashboard-page/blog-dashboard-page.constants';
+interface BlogDashboardBackendResponse {
+  'username': string;
+  'profile_picture_data_url': string;
+  'no_of_published_blog_posts': number;
+  'no_of_draft_blog_posts': number;
+  'published_blog_post_summary_dicts': BlogPostSummaryBackendDict[];
+  'draft_blog_post_summary_dicts': BlogPostSummaryBackendDict[];
+}
+
+interface NewBlogPostBackendResponse {
+  'blog_post_id': string
+}
+export interface BlogDashboardData {
+  username: string;
+  profilePictureDataUrl: string;
+  numOfPublishedBlogPosts: number;
+  numOfDraftBlogPosts: number;
+  publishedBlogPostSummaryDicts: BlogPostSummary[];
+  draftBlogPostSummaryDicts: BlogPostSummary[]
+}
```

Comment:
I think these should be in a separate typings file.

Reply:
Done.

Reply:
Reverted changes